### PR TITLE
Remove special handling of loss in progress bar

### LINKED ISSUE #9372: [RFC] Remove `{running,accumulated}_loss`
## Proposed refactoring or deprecation

Remove the following code: https://github.com/PyTorchLightning/pytorch-lightning/commit/a979944ea0e0b023e5a0a89b5695c259e61bd0cb

### Motivation

The running loss is a running window of loss values returned by the `training_step`. It has been present since the very beginning of Lightning and has become legacy code.

Problems:
- Users are sometimes confused by the value when they don't know it's a running window and compare it to the actual `loss` value they `self.log`ed.
- Often users `self.log` their actual loss which makes them see two "loss" values in the progress bar. 
- To disable it, you have to override the `get_progress_bar_dict` hook which is inconvenient.
- The running window configuration is opaque to the user as it's [hard-coded in the `TrainingBatchLoop.__init__`](https://github.com/PyTorchLightning/pytorch-lightning/blob/f239b9632017b64220f585e92f95e50c44445abf/pytorch_lightning/loops/batch/training_batch_loop.py#L38).

Alternative:

- Can be entirely replaced by asking users to call `self.log("loss", loss, prog_bar=True)`
- If the users still want to keep the "value window" functionality, this could be done by logging a `torchmetrics.Metric` specialized for it. (is there a Metric to replace the [`TensorRunningAccum`](https://github.com/PyTorchLightning/pytorch-lightning/blob/f239b9632017b64220f585e92f95e50c44445abf/pytorch_lightning/trainer/supporters.py#L39) already? cc @justusschock @awaelchli @akihironitta @rohitgr7 @SeanNaren @kaushikb11 @SkafteNicki)

### Pitch

Remove the code, I don't think there's anything to deprecate here.

- `get_progress_bar_dict` stays for the `v_num` and `split_idx`.
- The `TrainingBatchLoop.{accumulated,running}_loss` attributes should be private.
- The `FitLoop.running_loss` property seems to be there only for the `Tuner` and could be considered private: https://grep.app/search?q=fit_loop.running_loss
- No project seems to be using the `TensorRunningAccum`: https://grep.app/search?q=TensorRunningAccum

cc @awaelchli @ananthsub 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

### LINKED ISSUE #15831: Why is the loss in the progress bar different from the one logged manually
I follow the tutorial Lightning in 15 minutes(https://pytorch-lightning.readthedocs.io/en/stable/starter/introduction.html#maximize-flexibility).

But I don't understand what the training pbar means. Why is loss different from self.log("train_loss", loss, prog_bar=True)?

I look forward to receiving answers soon. Thanks!
图片
图片

cc @awaelchli

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #9372
Fixes #15831
Fixes issue 2 in https://github.com/Lightning-AI/lightning/issues/15884

cc @borda @carmocca 

### Does your PR introduce any breaking changes? If yes, please list them.

- Removed loop.running_loss
- Removed loop.accumulated_loss 
- Removed TensorRunningAccum class
- Removed "loss" item being logged by default in progress bar

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_fabric/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,13 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 
+## [2.0.0] - 202Y-MM-DD
+
+### Removed
+
+- Removed moving average of training step loss value for the progress bar ([#16192](https://github.com/Lightning-AI/lightning/issues/16192))
```
Comment (carmocca) [inline src/lightning_fabric/CHANGELOG.md:12]: This should be in the pl CHANGELOG

File: src/pytorch_lightning/callbacks/progress/base.py [hunk 2]
```diff
@@ -267,17 +267,20 @@ def get_standard_metrics(trainer: "pl.Trainer", pl_module: "pl.LightningModule")
...
+    loss_value = None
+    loss_metric = None
+
     # call .item() only once but store elements without graphs
-    running_train_loss = trainer.fit_loop.running_loss.mean()
-    avg_training_loss = None
-    if running_train_loss is not None:
-        avg_training_loss = running_train_loss.cpu().item()
+    if trainer.training:
+        loss_metric = trainer.fit_loop._results.get("training_step.train_loss")
```
Comment (carmocca) [inline src/pytorch_lightning/callbacks/progress/base.py:275]: Why keep this? Getting this key is arbitrary, users could log it with `tr_loss`, or `loss`, or `perdida` :es: 

We could use the return value of the `training_step`, but I suggest removing this logic completely.

**Review by carmocca — COMMENTED** [review]
The description says "part of https://github.com/Lightning-AI/lightning/issues/9372". Doesn't this close it? What would be left?

**Comment by awaelchli** [comment]
> The description says "part of https://github.com/Lightning-AI/lightning/issues/9372". Doesn't this close it? What would be left?

In this PR, I only intend to remove the moving average. The other item in the issue is about removing the loss from the progress bar. I think it's not something I want to do, maybe someone else could do it.

File: src/pytorch_lightning/callbacks/progress/base.py (hunk 2 above)
Comment (awaelchli) [reply src/pytorch_lightning/callbacks/progress/base.py:275]: Do you know why sometimes the value of `fit_loop._results.get("training_step.train_loss")` is of type `_ResultMetricCollection` and sometimes `_ResultCollection`?

**Comment by carmocca** [comment]
> I think it's not something I want to do, maybe someone else could do it.

We are kind of "forced" to do it, because we cannot assume the key `train_loss` to get a loss as I commented above.

**Review by carmocca — APPROVED** [review]

**Review by justusschock — APPROVED** [review]
